### ISSUE TITLE
Error: This update only applies to machines with Windows Subsystem for Linux

### ISSUE THREAD

**Issue by cnotti** [MicrosoftDocs/WSL#616]
I executed the wsl_update_x64 file without following any of the additional steps required to update from wsl to wsl2. It seemed to run without any issues, however, as I had not completed the other required steps, it didn't upgrade me to wsl2. 

Now when executing the wsl_update_x64 file after completing the additional required steps to upgrade, I get the error: "This update only applies to machines with Windows Subsystem for Linux" and I am stuck with version 1.

Any advise on how to get the file to execute under these circumstances?

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5d720f69-aff2-38d8-f0c6-9909e94acaa2
* Version Independent ID: 9d121625-e741-6d20-f3de-201a939290b8
* Content: [Updating the WSL 2 Linux kernel](https://docs.microsoft.com/en-nz/windows/wsl/wsl2-kernel#feedback)
* Content Source: [WSL/wsl2-kernel.md](https://github.com/MicrosoftDocs/WSL/blob/live/WSL/wsl2-kernel.md)
* Product: **dev-environment**
* Technology: **windows-subsystem-for-linux**
* GitHub Login: @craigloewen-msft
* Microsoft Alias: **crloewen**

**Comment by baonguyenhcm** [comment]
Just go to "Apps & features", scroll down until you see "Windows subsystem for Linux Update".
Uninstall it then re-install.

**Comment by cnotti** [comment]
I have just got it working by running wsl_update_x64 from powershell with admin rights as opposed to double clicking.

**Comment by Pearseak** [comment]
It won't prompt for elevated permissions,need to be rum from elevated cli

**Comment by ACure4IT** [comment]
Ensure to run wsl_update_x64.msi as an admin!

**Comment by alexsvt2** [comment]
@baonguyenhcm Thanks, now is working!

**Comment by noseratio** [comment]
I eventually got it installed by running: 

    wsudo msiexec -fav wsl_update_x64.msi

Disclaimer: I'm the author of [`wsudo`](https://github.com/noseratio/choco/tree/master/wsudo).

Thanks @craigloewen-msft for bringing me here. Would https://github.com/microsoft/WSL be a more proper to report issues?

**Comment by sonook** [comment]
Here the fix for the issue.

https://youtu.be/4czn-4uu65c

**Comment by ShubhamCanMakeCommit** [comment]
Restarting my machine solved the issue.

**Comment by Akela1101** [comment]
For me the solution was to go to "Apps & Features" / "Programs & Features" / "Turn Windows features on or off", and check the "Windows Subsystem for Linux".

Somehow that setting was unchecked.

**Comment by jonatansmith** [comment]
> Restarting my machine solved the issue.

Resolved for me too. Thanks for this comment.

**Comment by robertomarvin123** [comment]
Reinicie e deu certo!

> Reiniciar minha máquina resolveu o problema.

**Comment by ngabipoly** [comment]
This resolved it for me

**Comment by dyener** [comment]
Start-Process .\wsl_update_x64.msi
 
 This resolved it for me

**Comment by ekosis** [comment]
> In PowerShell Start-Process .\wsl_update_x64.msi
> 
> This resolved it for me

OMG.... working.... thank you

**Comment by Wu-yikun** [comment]
This blog may help you: https://blog.csdn.net/qq_39611230/article/details/117362198
but it can't help me 😭

**Comment by GHCMelo** [comment]
> In PowerShell Start-Process .\wsl_update_x64.msi
> 
> This resolved it for me

OMG! I've tried so many things, but only this worked!

**Comment by limabrunoo** [comment]
> In PowerShell Start-Process .\wsl_update_x64.msi
> 
> This resolved it for me

I tried this with admin rights on PowerShell and it worked for me! 
Thaks!

**Comment by SidinGit** [comment]
1. open PowerShell as admin 
2. navigate to ur downloads folder where the MSI file is present
3. type "msiexec /a wsl_update_x64.msi" . Here /a runs the file with admin access
4. Follow me on GitHub

**Comment by HugoRebelo99** [comment]
I was able to fix this issue by using wsl.exe --set-default-version 1, and then wsl.exe --install <distro>. After this it worked like a charm, I was able to install the distros but I was limited to version 1, every time I wanted to wsl.exe --set-version <distro> 2 it gave me a different error. WSL version 1 doesn't allow me to run Docker from within WSL. After a long time and checking other issues, I tried to update my WSL kernel and even this was unsuccessful. In the end, the folder C:\Windows\System32\lxss didn't contain the lib folder and the kernel and license files under C:\Windows\System32\lxss\tools. I just grabbed those files from another WSL installation on another PC and got the issue fixed! 😁
I don't know how those files went missing, but I assure you that disabling/enabling the Windows features such as Hyper-V, WSL, Windows Hypervisor Platform, and Virtual Machine Platform doesn't fix the issue nor bring these files back.

Hope this helps. 😉

TLDR: Check if you have C:\Windows\System32\lxss\tools\(kernel/license) files and if not try grabbing them from another WSL installation into your PC.

**Comment by RaunakShahi007** [comment]
> Downloading msi from https://github.com/microsoft/WSL/releases solved my problem.
